### PR TITLE
Use election2 instead of legacy election interface

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
Part of #1191

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
PTAL. Commit by commit review might be the best way.

File: server/trillian_log_signer/main.go [hunk 1]
```diff
@@ -66,10 +68,9 @@ var (
 		"Increase factor for tokens replenished by sequencing-based quotas (1 means a 1:1 relationship between sequenced leaves and replenished tokens)."+
 			"Only effective for --quota_system=etcd.")
 
-	preElectionPause    = flag.Duration("pre_election_pause", 1*time.Second, "Maximum time to wait before starting elections")
-	masterCheckInterval = flag.Duration("master_check_interval", 5*time.Second, "Interval between checking mastership still held")
```
Comment (pavelkalinnikov) [inline server/trillian_log_signer/main.go]: Need to add deletion of this flag to changelog as well.

File: util/election/runner.go [hunk 2]
```diff
@@ -105,80 +99,74 @@ func NewRunner(id string, cfg *RunnerConfig, tracker *MasterTracker, cancel cont
...
-				break  // no longer master
-			}
+		if err := er.beMaster(ctx, pending); err != nil {
+			glog.Errorf("%s: %v", er.id, err)
+			break
 		}
 	}
 }
 
+func (er *Runner) beMaster(ctx context.Context, pending chan<- Resignation) error {
```
Comment (gdbelvin) [inline util/election/runner.go]: Thanks for splitting this out into a separate function

File: util/election/runner_test.go [hunk 3]
```diff
@@ -59,91 +59,101 @@ func TestConfigResignDelay(t *testing.T) {
...
-			var wg sync.WaitGroup
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
 			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
 
-			startTime := time.Now()
-			fakeTimeSource := clock.NewFake(startTime)
+			e := to.NewElection()
+			d := to.NewDecorator(e)
```
Comment (gdbelvin) [inline util/election/runner_test.go]: If you don't use e anywhere, you might be able to nest
`d := to.NewDecorator(to.NewElection())`

File: server/trillian_log_signer/main.go (hunk 1 above)
Comment (pavelkalinnikov) [reply server/trillian_log_signer/main.go]: Done.

File: util/election/runner_test.go (hunk 3 above)
Comment (pavelkalinnikov) [reply util/election/runner_test.go]: It's used one time in line 134. I could use `d` there, but I would like that line to to ignore any `Resign` errors in the decorator, to simulate mastership loss properly.

**Comment by pavelkalinnikov** [comment]
The containers don't use the flag luckily.

**Pull request merged**